internal/backend/backend.go

```diff
@@ -26,10 +26,52 @@ import (
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
 
 	"github.com/snyk/kubernetes-scanner/internal/config"
 )
 
+// the default transport automatically honors HTTP_PROXY settings.
+// this value is overwritten by init with a roundtripper that has prometheus metrics.
+var transport = http.DefaultTransport
+
+func init() {
+	commonLabels := []string{"code", "method"}
+	// all supported metrics:
+	var (
+		requests = prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Subsystem: "http_outgoing",
+				Name:      "requests_total",
+				Help:      "A counter for outgoing requests.",
+			},
+			commonLabels,
+		)
+
+		durations = prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{
+				Subsystem: "http_outgoing",
+				Name:      "request_duration_histogram_seconds",
+				Help:      "Request time duration.",
+				Buckets:   prometheus.DefBuckets,
+			},
+			commonLabels,
+		)
+	)
+
+	metrics.Registry.MustRegister(&httpMetricsCollector{
```

Comment:
From the PR description:

> Not perfect, but simple and probably good enough for a long time

What are the downsides? This seems like a fine approach to me.

If you want to consider an alternative, there is also `Manager.AddMetricsExtraHandler()` in controller-runtime, which allows mounting an extra promhttp handler on a different http path. Users would have to configure collection for that path too, which we could add to the ServiceMonitor, but unless there's an issue with this approach, I wouldn't bother 👍

Reply:
Just using `init` and global vars. "Perfect" code doesn't use that 😄 
But *working* code does, so I'd also say it's okay ;) 

RE the handler, yes, but i also don't see this making sense either. The current approach is good enough 😉